### PR TITLE
resource/rds_cluster_instance: Support availability_zone

### LINKED ISSUE #2642: Feature request: support specifying availbility_zone for aws_rds_cluster_instance
Hi, 

It seems like we cannot specify `availbility_zone` when creating Aurora instances, per [documentation](https://www.terraform.io/docs/providers/aws/r/rds_cluster_instance.html). It would be super helpful to have this option because it's available on other kinds of RDS instances.

Thanks,
Ruobing

### PR DESCRIPTION
**Pull request by atsushi-ishibashi**
Closes #2642

### CODE REVIEW

**Comment by atsushi-ishibashi** [comment]
If I use `data "aws_availability_zones"`, it seemed to be difficult to check resource attribute so I specified the constant.
```
TF_ACC=1 go test ./aws -v -run=TestAccAWSRDSClusterInstance_az -timeout 120m
=== RUN   TestAccAWSRDSClusterInstance_az
--- PASS: TestAccAWSRDSClusterInstance_az (799.15s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	799.194s
```

**Comment by radeksimko** [comment]
@atsushi-ishibashi Thanks for the update.

I understand it's difficult to reference the resource attribute, but the problem is that this test is now pinned to `us-west-2` and more importantly may not run in all accounts as AWS often assigns different AZ names to different accounts.

How do you feel about using the AZ data source? e.g.

```hcl
data "aws_availability_zones" "available" {}

resource "aws_rds_cluster" "default" {
...
  availability_zones = ["${data.aws_availability_zones.available.names}"]
...
}

resource "aws_rds_cluster_instance" "cluster_instances" {
...
  availability_zone       = "${data.aws_availability_zones.available.names[0]}"
...
```

**Comment by atsushi-ishibashi** [comment]
Sure👍
But how should I write the below line? @radeksimko 
https://github.com/terraform-providers/terraform-provider-aws/pull/2812/files#diff-e2a347e0fe3dfd86606d211eb111c503R66

The main purpose of this test is to confirm whether the specified az is set as availability_zone.

File: aws/resource_aws_rds_cluster_instance_test.go [hunk 1]
```diff
@@ -59,11 +59,11 @@ func TestAccAWSRDSClusterInstance_az(t *testing.T) {
 		CheckDestroy: testAccCheckAWSClusterDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccAWSClusterInstanceConfig(acctest.RandInt()),
+				Config: testAccAWSClusterInstanceConfig_az(acctest.RandInt()),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckAWSClusterInstanceExists("aws_rds_cluster_instance.cluster_instances", &v),
 					testAccCheckAWSDBClusterInstanceAttributes(&v),
-					resource.TestCheckResourceAttr("aws_rds_cluster_instance.cluster_instances", "availability_zone", "us-west-2a"),
+					//resource.TestCheckResourceAttr("aws_rds_cluster_instance.cluster_instances", "availability_zone", "us-west-2a"),
```
Comment (radeksimko) [inline aws/resource_aws_rds_cluster_instance_test.go]: It's ok to check this more loosely, e.g. 

```go
resource.TestMatchResourceAttr("aws_rds_cluster_instance.cluster_instances", "availability_zone", regexp.MustCompile("^us-west-2[a-z]{1}$")
```

Comment (atsushi-ishibashi) [reply aws/resource_aws_rds_cluster_instance_test.go]: I got it👍

**Comment by atsushi-ishibashi** [comment]
@radeksimko Updated👍
```
TF_ACC=1 go test ./aws -v -run=TestAccAWSRDSClusterInstance_az -timeout 120m
=== RUN   TestAccAWSRDSClusterInstance_az
--- PASS: TestAccAWSRDSClusterInstance_az (920.21s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	**920.255s**
```

**Comment by bflad** [comment]
This has been released in [version 1.10.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1100-february-24-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**